iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -363,14 +363,16 @@ public async Task RetrySetRetryPolicyVerifyInternalsSuccess()
                });
 
             // act and assert
-            await sut.OpenAsync(CancellationToken.None).ExpectedAsync<IotHubClientException>().ConfigureAwait(false);
+            var exception = await sut.OpenAsync(CancellationToken.None).ExpectedAsync<IotHubClientException>().ConfigureAwait(false);
+            exception.StatusCode.Should().Be(IotHubStatusCode.NetworkErrors);
             nextHandlerCallCounter.Should().Be(2);
             retryPolicy.Counter.Should().Be(2);
 
             var noretry = new NoRetry();
             sut.SetRetryPolicy(noretry);
-            await sut.OpenAsync(CancellationToken.None).ExpectedAsync<IotHubClientException>().ConfigureAwait(false);
 
+            exception = await sut.OpenAsync(CancellationToken.None).ExpectedAsync<IotHubClientException>().ConfigureAwait(false);
+            exception.StatusCode.Should().Be(IotHubStatusCode.NetworkErrors);
```

Comment:
Should the expected error code be ```Timeout``` now?

Reply:
No. This test is intended to throw `IotHubClientException` with `NetworkErrors` when opening the RetryDelegatingHandler to mimic the transient network errors, so `NetworkErrors` is expected.